backend/core/src/main/java/org/sonarsource/sonarlint/core/repository/rules/RulesRepository.java

```diff
@@ -100,6 +104,18 @@ public void connectionRemoved(ConnectionConfigurationRemovedEvent e) {
     evictAll(e.getRemovedConnectionId());
   }
 
+  @EventListener
+  public void configScopeRemoved(ConfigurationScopeRemovedEvent e) {
+    var removedBindingConfiguration = e.getRemovedBindingConfiguration();
+    if (removedBindingConfiguration.isBound() && hasNoMoreBindings(removedBindingConfiguration.getConnectionId())) {
```

Comment:
nit pick: we can extract `removedBindingConfiguration.getConnectionId()` to a var as it is used on two places